average_calculation.py

```diff
@@ -24,29 +24,63 @@
 # the rest of them is optional
 
 api = RiaAPI()
+search_params = {
+    'category': 'Легковые',
+    'mark': 'Mitsubishi',
+    'model': 'ASX',
+    # 'bodystyle': '',
+    'years': [2015, 2017],
+    # 'state': 'Винницкая',
+    # 'city': 'Винница',
+    'gears': ['Автомат'],
+    # 'opts': ['ABS'],
+    # 'mileage': [10, 200],
+    # 'fuels': ['Дизель'],
+    # 'drives': ['Полный'],
+    # 'color': 'Серый',
+    # 'engine_volume': 1.5,
+    # 'seats': 5,
+    # 'doors': 3,
+    # 'carrying': 1500,
+    'custom': False,
+    'damage': False,
+    'under_credit': False,
+    'confiscated': False,
+    'on_repair_parts': False
+}
 myCarAveragePrice = RiaAverageCarPrice(
-    category='Легковые',
-    mark='Renault',
-    model='Sceni',
-    # bodystyle=None,
-    # years=[2006, 2007],
-    state='Винницкая',
-    city='Винница',
-    # gears=['Механика'],
-    # opts=['ABS'],
-    # mileage=[10, 200],
-    # fuels=['Дизель'],
-    # drives=['Полный'],
-    # color='Серый',
-    # engine_volume=1.5,
-    # seats=5,
-    # doors=3,
-    # carrying=1500,
-    custom=False,
-    damage=False,
-    under_credit=False,
-    confiscated=False,
-    on_repair_parts=False
+    category=search_params.get('category'),
+    mark=search_params.get('mark'),
+    model=search_params.get('model'),
+    bodystyle=search_params.get('bodystyle'),
+    years=search_params.get('years'),
+    state=search_params.get('state'),
+    city=search_params.get('city'),
+    gears=search_params.get('gears'),
+    opts=search_params.get('opts'),
+    mileage=search_params.get('mileage'),
+    fuels=search_params.get('fuels'),
+    drives=search_params.get('drives'),
+    color=search_params.get('color'),
+    engine_volume=search_params.get('engine_volume'),
+    seats=search_params.get('seats'),
+    doors=search_params.get('doors'),
+    carrying=search_params.get('carrying'),
+    custom=search_params.get('custom'),
+    damage=search_params.get('damage'),
+    under_credit=search_params.get('under_credit'),
+    confiscated=search_params.get('confiscated'),
+    on_repair_parts=search_params.get('on_repair_parts')
 )
 
-pprint(myCarAveragePrice.get_average())
+base_url = "https://auto.ria.com/auto__{}.html"
+average = myCarAveragePrice.get_average()
+
+report = {
+    'search_params': search_params,
+    'average': average,
+    'classifieds_urls': [
+        base_url.format(item)
+        for item in average.get('classifieds')]
+}
+pprint(report)
```

Comment:
And this looks even nicer:

```
import json

print(json.dumps(report, indent=2))
```

Reply:
I like more the one with `pprint()` :sweat_smile: